doc/sphinx/users-guide/vcl-grace.rst

```diff
@@ -19,52 +23,194 @@ If you are serving thousands of hits per second the queue of waiting
 requests can get huge. There are two potential problems - one is a
 thundering herd problem - suddenly releasing a thousand threads to
 serve content might send the load sky high. Secondly - nobody likes to
-wait. To deal with this we can instruct Varnish to keep
-the objects in cache beyond their TTL and to serve the waiting
-requests somewhat stale content.
+wait.
+
+Setting an object's `grace` to a positive value tells Varnish that it
+should serve the object to clients for some time after the TTL has
+expired, while Varnish fetches a new version of the object. The default
+value is controlled by the runtime parameter ``default_grace``.
+
+Keep
+~~~~
+
+Setting an object's `keep` tells Varnish that it should keep an object
+in the cache for some additional time. There are two reasons to do this:
 
-So, in order to serve stale content we must first have some content to
-serve. So to make Varnish keep all objects for 2 minutes beyond their
-TTL use the following VCL::
+* To use the object as a candidate for ``304 NOT MODIFIED`` from the server.
```

Comment:
more precise:
* To use the object to construct a conditional ``GET`` backend request (with ``If-Modified-Since:`` and/or ``Ìf-None-Match:`` headers), allowing the backend to reply with a ``304 Not Modified`` response, which may be more efficient on the backend and saves re-transmitting the unchanged body.

Reply:
Excellent, thanks.